### PR TITLE
Document (non-contentious parts of) Sass settings layer

### PR DESCRIPTION
**Pull request by 36degrees**
This is the first step in clarifying what comprises the public API of Frontend.

This adds Sassdocs for the following files:

- `settings/_assets.scss`
- `settings/_colours-applied.scss`
- `settings/_compatibility.scss`
- `settings/_global-styles.scss`
- `settings/_ie8.scss`
- `settings/_measurements.scss`
- `settings/_media-queries.scss`
- `settings/_typography-font-families.scss` (renamed from `_typography-font-stacks.scss`)
- `settings/_typography-font.scss`

That leaves some files in settings which are not using Sassdoc:

- `settings/_colours-organisations.scss`
- `settings/_colours-palette.scss`
- `settings/_spacing.scss`
- `settings/_typography-responsive.scss`

I've left these (for now) because they all contain a large number of variables, and we've discussed moving them to maps and adding helper functions which will give us more control over e.g. deprecating colours.

As part of these changes we've also:

- Defined private or public access for all settings
- Prefixed private variables with an underscore
- Made settings `!default` where it makes sense to do so
- Moved the button variables into the button component
- Refered to font families rather than font stacks consistently
- Renamed nta-light to font-family-nta
- Correctly used abstracted `$govuk-font-family-tabular` for tabular number mixins, rather than referencing nta-light-tabular directly
- Abstracted away `sass-mq`, and expose our own settings in our public api

https://trello.com/c/B8OCSU8R/759-agree-on-and-document-public-api-for-govuk-frontend

### CODE REVIEW

File: src/helpers/grid.test.js [hunk 1]
```diff
@@ -115,7 +119,7 @@ describe('grid system', () => {
           box-sizing: border-box;
           width: 100%;
           padding: 0 15px; }
-          @media (min-width: 46.25em) {
+          @media (min-width: 40.0625em) {
```
Comment (alex-ju) [inline src/helpers/grid.test.js]: What's the reason behind the change of these values?

**Comment by igloosi** [comment]
this looks solid, happy to approve it now, but i guess others should have a look ( as they've been involved in the discussion)

File: src/helpers/grid.test.js (hunk 1 above)
Comment (36degrees) [reply src/helpers/grid.test.js]: Good question - it used to use the default `sass-mq` breakpoints, now it's using the ones defined in our settings.

Comment (alex-ju) [reply src/helpers/grid.test.js]: Could we add this to the PR comments?

Comment (36degrees) [reply src/helpers/grid.test.js]: Done.

**Comment by 36degrees** [comment]
Argh, CHANGELOG.
